### PR TITLE
operations on predicates

### PR DESCRIPTION
**Pull request by eparejatobes**

### CODE REVIEW

File: src/test/scala/cosas/KListsTests.scala [hunk 1]
```diff
@@ -376,11 +376,22 @@ class KListTests extends org.scalatest.FunSuite {
...
     assert { isAnyVal('x') === True }
+    assert { isInt("foo") === False }
 
     assertResult('b' :: true :: 2 :: 'a' :: *[Any]) {
       ('b' :: true :: "hola" :: 2 :: 'a' :: *[Any]).filter(isAnyVal)
     }
+
+    assertResult(2 :: *[Any]) {
+      ('b' :: Set("a") :: true :: "hola" :: List(1,2,3) :: 2 :: 'a' :: *[Any]).filter(isAnyVal ∧ isInt)
+    }
```
Comment (laughedelic) [inline src/test/scala/cosas/KListsTests.scala]: nice :+1:

Comment (eparejatobes) [inline src/test/scala/cosas/KListsTests.scala]: yes, and it was simpler than I expected

**Comment by eparejatobes** [comment]
by the way, why we need a predicate type?

**Comment by laughedelic** [comment]
> by the way, why we need a predicate type?

for filtering and stuff...

**Comment by eparejatobes** [comment]
because of the false-by-default thing I guess? bu that could work for any depfn with out Bool.

**Comment by laughedelic** [comment]
and what's the point of using arbitrary depfns for predicates? it's like when you have a implicit cast to Bool: everything except of zero is true :trollface:

**Comment by eparejatobes** [comment]
OK fine fair enough. Let's add syntax for transforming a depfn1 with out Bool to a predicate

**Comment by eparejatobes** [comment]
done. Would you like it to be an implicit conversion?

**Comment by laughedelic** [comment]
No. I think explicitness is good here. I would rather make a syntax method: `trueOnLists.asPredicates`. But it's not crucial.

**Comment by eparejatobes** [comment]
done. Merging.

**Pull request merged**